src/ATen/native/xpu/sycl/UpSampleLinear1dKernels.cpp

```diff
@@ -0,0 +1,222 @@
+        using KernelClass =
+            UpsampleLinear1dKernelFunctor<scalar_t, accscalar_t>;
+        int64_t local_range = syclMaxWorkGroupSize<KernelClass>();
+        KernelClass kfn(num_kernels, rwidth, align_corners, idata, odata);
+        auto global_range = (num_kernels + local_range - 1) / local_range;
+        sycl_kernel_submit(
+            global_range * local_range,
+            local_range,
+            getCurrentSYCLQueue(),
+            kfn);
+      });
+}
+
+template <typename scalar_t, typename accscalar_t>
+struct UpsampleLinear1dBackwardKernelFunctor {
+  void operator()(sycl::nd_item<1> item) const {
+    int index =
+        item.get_local_id(0) + item.get_group(0) * item.get_local_range(0);
+
+    const int batchsize = idata_.size(0);
+    const int channels = idata_.size(1);
+    const int width1 = idata_.size(2);
+    const int width2 = odata_.size(2);
+    PackedTensorAccessor64<scalar_t, 3> idata_res = idata_;
+    if (index < n_) {
+      const int w2 = index % width2;
+      if (width1 == width2) {
+        const int w1 = w2;
+        for (int n = 0; n < batchsize; n++) {
+          for (int c = 0; c < channels; ++c) {
+            const scalar_t val = odata_[n][c][w1];
+            idata_res[n][c][w2] = val;
+          }
+        }
+        return;
+      }
+      const accscalar_t w1r = area_pixel_compute_source_index<accscalar_t>(
+          rwidth_, w2, align_corners_, /*cubic=*/false);
+      const int w1 = w1r;
+      const int w1p = (w1 < width1 - 1) ? 1 : 0;
+      const accscalar_t w1lambda = w1r - w1;
+      const accscalar_t w0lambda = static_cast<accscalar_t>(1) - w1lambda;
+
+      for (int n = 0; n < batchsize; n++) {
+        for (int c = 0; c < channels; ++c) {
+          const scalar_t d2val = odata_[n][c][w2];
+          atomicAdd(
+              (sycl_global_ptr<scalar_t>)(&idata_res[n][c][w1]),
+              static_cast<scalar_t>(w0lambda * d2val));
+          atomicAdd(
+              (sycl_global_ptr<scalar_t>)(&idata_res[n][c][w1 + w1p]),
+              static_cast<scalar_t>(w1lambda * d2val));
+        }
+      }
+    }
+  }
+  UpsampleLinear1dBackwardKernelFunctor(
+      const int n,
+      const accscalar_t rwidth,
+      const bool align_corners,
+      PackedTensorAccessor64<scalar_t, 3> idata,
+      const PackedTensorAccessor64<const scalar_t, 3> odata)
+      : n_(n),
+        rwidth_(rwidth),
+        align_corners_(align_corners),
+        idata_(idata),
+        odata_(odata) {}
+
+ private:
+  const int n_;
+  const accscalar_t rwidth_;
+  const bool align_corners_;
+  PackedTensorAccessor64<scalar_t, 3> idata_;
+  const PackedTensorAccessor64<const scalar_t, 3> odata_;
+};
+
+void upsample_linear1d_backward_kernel(
+    const Tensor& grad_output_,
+    IntArrayRef output_size,
+    IntArrayRef input_size,
+    bool align_corners,
+    std::optional<double> scales,
+    Tensor& grad_input) {
+  globalContext().alertNotDeterministic("upsample_linear1d_backward_out_xpu");
```

Comment:
Does CUDA has deterministic implementation?

Reply:
`upsample_linear1d_backward_xpu`

Reply:
CUDA is not deterministic.